### PR TITLE
update CHANGELOG.md

### LINKED ISSUE #11167: Update release note in the TiKV repo
The CHANGELOG.md in the TiKV repo has been not changed for months.

The changelog file is useful when people want to search for when a feature is introduced, or when a bug is resolved.

### PR DESCRIPTION
**Pull request by xiejiandong**
### What is changed and how it works?
Update CHANGELOG.md to 20220105;
Close #11167 

What's Changed:
- PR to update `tikv/CHANGELOG.md`:
- PR update content since tikv-server v5.0.1(20210423);
- PR update content refers to "https://github.com/tikv/tikv/releases"

### Check List
Tests - Preview works good.
No code

### Release note
```release-note
None
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +203,77 @@ See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.
...
+  + Reduce the size of Resolved TS messages to save network bandwidth [#2448](https://github.com/tikv/tikv/pull/2448)
++ Bug fixes
+  + Fix the issue that BR reports the "file already exists" error when TDE is enabled during data restore [#10917](https://github.com/tikv/tikv/pull/10917)
+  + Fix the issue that TiKV deletes stale Regions too frequently [#10781](https://github.com/tikv/tikv/pull/10781)
+  + Fix the issue that TiKV frequently reconnects the PD client [#9818](https://github.com/tikv/tikv/pull/9818)
+  + Check stale file information from the encryption file dictionary [#10598](https://github.com/tikv/tikv/pull/10598)
+
+## [4.0.14] - 2021-07-27
+
++ Compatibility changes
```
Comment (tabokie) [inline CHANGELOG.md:238]:
```suggestion
+ Compatibility Changes
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +203,77 @@ See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.
...
+  + Change the default value of merge-check-tick-interval from 10 to 2 to speed up the Region merge process [#9676](https://github.com/tikv/tikv/pull/9676)
++ Feature enhancements
```
Comment (tabokie) [inline CHANGELOG.md:240]: ditto

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,182 @@
...
+  + Fix the issue that batch messages are too large in Raft client implementation [#9714](https://github.com/tikv/tikv/pull/9714)
+  + Fix the issue that some coroutines leak in resolved_ts [#10965](https://github.com/tikv/tikv/pull/10965)
+  + Fix a panic issue that occurs to the coprocessor when the size of response exceeds 4 GiB [#9012](https://github.com/tikv/tikv/pull/9012)
+  + Fix the issue that snapshot Garbage Collection (GC) misses GC snapshot files when snapshot files cannot be garbage collected [#10813](https://github.com/tikv/tikv/pull/10813)
+  + Fix a panic issue caused by timeout when processing Coprocessor requests [#10852](https://github.com/tikv/tikv/pull/10852)
+  + Fix a memory leak caused by monitoring data of statistics threads [#11195](https://github.com/tikv/tikv/pull/11195)
+  + Fix a panic issue caused by getting the cgroup information from some platforms [#10980](https://github.com/tikv/tikv/pull/10980)
+
+## [5.2.3] - 2021-12-02
++ Bug fix
```
Comment (tabokie) [inline CHANGELOG.md:26]:
```suggestion
## [5.2.3] - 2021-12-02

+ Bug Fixes
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -27,6 +203,77 @@ See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.
...
+  + Fix the issue of negative sign when the decimal divide result is zero [#29586](https://github.com/tikv/tikv/pull/29586)
+  + Fix the issue that the average latency of the by-instance gRPC requests is inaccurate in TiKV metrics [#11299](https://github.com/tikv/tikv/pull/11299)
+  + Fix the issue of TiCDC panic that occurs when the downstream database is missing [#11123](https://github.com/tikv/tikv/pull/11123)
+  + Fix the issue that the Raft connection is broken when the channel is full [#11047](https://github.com/tikv/tikv/pull/11047)
+  + Fix the issue that TiDB cannot correctly identify whether the Int64 types in Max/Min functions are a signed integer or not, which causes the wrong calculation result of Max/Min [#10158](https://github.com/tikv/tikv/pull/10158)
+  + Fix the issue that CDC adds scan retries frequently due to the Congest error [#11082](https://github.com/tikv/tikv/pull/11082)
+
+## [4.0.15] - 2021-09-23
+
++ Feature enhancement
```
Comment (tabokie) [inline CHANGELOG.md:223]: File it under improvements.

**Review by tabokie — COMMENTED** [review]
Could you use the title "Bug Fixes" regardless of the actual fix number? Rest looks good.

**Review by tabokie — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**